src/OrleansLoggerProvider/OrleansLoggingFactoryExtension.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Net;
+
+namespace Orleans.Extensions.Logging
+{
+    public static class OrleansLoggingFactoryExtension
+    {
+        /// <summary>
+        /// Add message bulking feature onto <param name="provider"></param>, and add that new logger provider 
+        /// <see cref="MessageBulkingLoggerProvider{TDecoratedLoggerProvider}"/> into <param name="builder"></param>.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="provider"></param>
+        /// <param name="messageBulkingConfig"></param>
+        /// <returns></returns>
+        public static ILoggingBuilder AddMesageBulkingLoggerProvider<TDecoratedLoggerProvider>(this ILoggingBuilder builder, TDecoratedLoggerProvider provider, MessageBulkingConfig messageBulkingConfig)
+            where TDecoratedLoggerProvider : ILoggerProvider
+        {
+            builder.AddProvider(new MessageBulkingLoggerProvider<TDecoratedLoggerProvider>(provider, messageBulkingConfig));
```

Comment:
this should probably be conditional on whether the bulking options are not null. Otherwise just add the other provider directly.

Reply:
I think the correct fix here is making bulkConfig optional, since it do have default config values. 
This extension method is designed to add event bulking feature onto the user's logger provider, so its behavior should stick with that IMO. 
If the user doesn't want event bulking feature, he/she can just use native builder.AddProvider(their logger provider).

Reply:
well, the point is that this is also being used by AddLegacyOrleansLogger... Then if not here, do the conditional there first and decide whether to call AddEventBulkingLoggerProvider whether that one is non-null